terraform/ec2/linux/main.tf

```diff
@@ -96,7 +96,10 @@ resource "null_resource" "integration_test" {
       "echo clone and install agent",
       "git clone --branch ${var.github_test_repo_branch} ${var.github_test_repo}",
       "cd amazon-cloudwatch-agent-test",
-      "aws s3 cp s3://${var.s3_bucket}/integration-test/binary/${var.cwa_github_sha}/linux/${var.arc}/${var.binary_name} .",
+      // If canary fetch the latest binary else fetch commit binary
+      var.is_canary ?
+      "aws s3 cp s3://${var.s3_bucket}/release/amazon_linux/${var.arc}/latest/${var.binary_name} ."
+      : "aws s3 cp s3://${var.s3_bucket}/integration-test/binary/${var.cwa_github_sha}/linux/${var.arc}/${var.binary_name} .",
```

Comment:
Looks fine to me.
But if we are passing the bucket to the `go test` command later on, then the test code could download the RPM instead of doing it here, right?